### PR TITLE
SIMD tests

### PR DESCRIPTION
**Pull request by ANDREENKOS**
Commit includes SIMD tests of two types:
1) Intrinsic tests, that call functions from RyuJIT SIMD intrinsic list.
2) Bytecode tests, which generate different bytecodes in IL with SIMD
struct.

### CODE REVIEW

**Comment by ANDREENKOS** [comment]
@dotnet-bot retest this please

File: tests/src/JIT/SIMD/AbsGeneric.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using System.Numerics;
+
+namespace VectorMathTests
+{
+    class Program
+    {
+
+
+
+        static int Main(string[] args)
```
Comment (jasonwilliams200OK) [inline tests/src/JIT/SIMD/AbsGeneric.cs]: nit: some files have extra whitespace while other `.cs` tests files under `JIT/` don't have extra linefeeds between `class name \n{` and the first method.

File: tests/src/JIT/SIMD/AbsGeneric.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            var c = new System.Numerics.Vector<long>(-11);
+            c = System.Numerics.Vector.Abs(c);
+            if (c[1] != 11)
+            {
+                return 0;
+            }
+
+            var d = new System.Numerics.Vector<double>(-100.0);
+            d = System.Numerics.Vector.Abs(d);
+            if (d[0] != 100)
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/AbsGeneric.cs]: You should never be checking floating point values for (in)equality. I would suggest using a CheckValue<T> method that does the appropriate check, and would also suggest that these tests could be made generic to reduce the amount of code and increase the coverage.

File: tests/src/JIT/SIMD/AbsGeneric.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                return 0;
+            }
+            var g = new System.Numerics.Vector<ulong>(21);
+            g = System.Numerics.Vector.Abs(g);
+            if (g[1] != 21)
+            {
+                return 0;
+            }
+
+            return 100;
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/AbsGeneric.cs]: I would prefer to see defined constants for Pass and Fail.

File: tests/src/JIT/SIMD/AbsSqrt.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            Point a = new Point(1, 2, 3, 4);
+            Point b = new Point(2, 2, 1, 1);
+            float c = 33;
+            Point d = (b + a) * c;
+            Point q = d + a;
+            Point s = Point.SquareRoot(q);
+            s *= -1;
+            s = Point.Abs(s);
+            return (int)(s).X * 10;
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/AbsSqrt.cs]: Please don't combine a computation with the Pass/Fail return value - it obscures what is going on. Rather, you should either add a comment that 100 is the expected value of the expression (and compare against that), or assign it to a variable such as "Result" and compare against the expected value. Then, return Pass or Fail.

File: tests/src/JIT/SIMD/Add.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    class Program
+    {
+
+      
+
+        static int Main(string[] args)
+        {
+			Point a = new Point(3, 60);
+			Point b = new Point(10, 40);
+            return (int)(a + b).Y;
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/Add.cs]: See above.

File: tests/src/JIT/SIMD/BitwiseOperations.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+        static byte NextByte(Random random)
+        {
+            return (byte)random.Next(0, 255);
+        }
+
+        static int TestBool()
+        {
+            Random random = new Random(13);
+            byte[] arr1 = new byte[] { NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random), NextByte(random) };
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/BitwiseOperations.cs]: Please add a "GetRandomArray<T>" or something like that.

File: tests/src/JIT/SIMD/CircleInConvex.cs [hunk 7]
```diff
@@ -0,0 +1,286 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Numerics;
+
+using Point = System.Numerics.Vector2;
+
+
+namespace ClassLibrary
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/CircleInConvex.cs]: The more complex tests here need comments.

File: tests/src/JIT/SIMD/CircleInConvex.cs [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+{
+
+    public class test
+    {
+        const float EPS = 1E-9F;
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/CircleInConvex.cs]: Should this be Single.Epsilon?

File: tests/src/JIT/SIMD/CircleInConvex.cs [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+        const int steps = 100;
+        const float INF = 1000000;
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/CircleInConvex.cs]: Single.PositiveInfinity?

Comment (CarolEidt) [inline tests/src/JIT/SIMD/CircleInConvex.cs]: (1000000 doesn't really seem like a reasonable value here)

Comment (CarolEidt) [inline tests/src/JIT/SIMD/CircleInConvex.cs]: Unless what you want is just "some large value" (not infinity)

File: tests/src/JIT/SIMD/Ldind.cs [hunk 10]
```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Point = System.Numerics.Vector4;
+
+
+namespace Test
+{
+    [StructLayout(LayoutKind.Explicit, Pack = 1, Size = 8)]
```
Comment (CarolEidt) [inline tests/src/JIT/SIMD/Ldind.cs]: Why is this here?

**Comment by CarolEidt** [comment]
@sivarv - if you have time could you also take a look?

File: tests/src/JIT/SIMD/AbsGeneric.cs (hunk 2 above)
Comment (mikedn) [inline tests/src/JIT/SIMD/AbsGeneric.cs]: In the particular case of `abs` I think that equality is warranted, I wouldn't expect `abs(-100.0)` to be something like 100.000000000001.

File: tests/src/JIT/SIMD/AbsGeneric.cs (hunk 3 above)
Comment (ANDREENKOS) [inline tests/src/JIT/SIMD/AbsGeneric.cs]: Can you show example? I didn't see using of such example in existing tests.
I saw test like 25param1a.cs.

Comment (CarolEidt) [inline tests/src/JIT/SIMD/AbsGeneric.cs]: The newer tests in CodeGenBringupTests have Pass and Fail defined.

**Comment by ANDREENKOS** [comment]
I updated tests, fixed formatting problems and deleted useless tests.

**Comment by jasonwilliams200OK** [comment]
:+1:

Please run 

```cmd
git remote add dotnet https://github.com/dotnet/coreclr
git checkout SIMDTests
git pull --rebase dotnet master
```

to avoid the merge commit (df87ec6) landing in this PR.

**Comment by pgavlin** [comment]
LGTM as well. Thanks for adding these.

**Comment by CarolEidt** [comment]
LGTM
